pkg/webhook/server/generic/collaset/collaset_mutating_handler.go

```diff
@@ -63,6 +67,25 @@ func (h *MutatingHandler) Handle(ctx context.Context, req admission.Request) (re
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
 
+	for _, podName := range cls.Spec.ScaleStrategy.PodToDelete {
+		pod := &corev1.Pod{}
+		if err := h.Client.Get(ctx, types.NamespacedName{Namespace: cls.Namespace, Name: podName}, pod); err != nil {
+			if errors.IsNotFound(err) {
+				return admission.Errored(http.StatusBadRequest, fmt.Errorf("no found Pod named %s to delete", podName))
+			}
+			return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failed to find Pod %s: %v", podName, err))
+		}
+		owned := false
+		for _, ref := range pod.OwnerReferences {
+			if ref.Name == cls.Name {
+				owned = true
+			}
+		}
+		if !owned {
+			return admission.Errored(http.StatusBadRequest, fmt.Errorf("not allwed to delete pod %s which is not owned by collaset %s", podName, cls.Name))
+		}
```

Comment:
This may be superfluous, as the Pod is deleted by the CollaSet controller, and thus it can no longer be guaranteed that the user will update `cls.Spec.ScaleStrategy.PodToDelete`.

Reply:
removed by https://github.com/KusionStack/operating/pull/199/commits/848891d5adbf7c4309c8ffa2fab1c9c67e9fc5f6